llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -4314,44 +4314,584 @@ AMDGPUInstructionSelector::selectVOP3NoMods(MachineOperand &Root) const {
   }};
 }
 
-std::pair<Register, unsigned>
-AMDGPUInstructionSelector::selectVOP3PModsImpl(
-  Register Src, const MachineRegisterInfo &MRI, bool IsDOT) const {
+enum class SrcStatus {
+  IS_SAME,
+  IS_UPPER_HALF,
+  IS_LOWER_HALF,
+  IS_UPPER_HALF_NEG,
+  IS_LOWER_HALF_NEG,
+  IS_HI_NEG,
```

Comment:
What's the difference between IS_UPPER_HALF_NEG and IS_HI_NEG? Add comments to explain what all of these mean.